### DISCUSSION TITLE
Task entry order issue, for tasks that provide for existing tasks

### DISCUSSION THREAD

**Discussion by primal-buddhist** [GothenburgBitFactory/taskwarrior#2528] [Q&A]
Hi
So, my task creation flow works by setting a goal, then brainstorming the next task that will get me there, sometimes several. 

eg

I create task 1 - Tidy the garage

Then I think, ah, need to get some bin bags

So I add task 2 which I now want to be in front of Task 1.

So now I have to then modify task 1 to set dep:2, so that's two commands. 

What would be better would have some sort of "predecessor" command which sets the dependency for the previously entered task.

eg

Task add Get bin bags pre:1

And now Task 1 has it's dependency set to 2 without two commands. 

Any way of doing this now?

**Comment by tbabej** [comment]
There is no way of doing that with vanilla taskwarrior currently. One available option, however, is to use this user-contributed hook,

https://gist.github.com/wbsch/a2f7264c6302918dfb30

which introduces the `blocks:` virtual attribute, that allows you to set dependencies in reverse:

```bash
task add Something
task add This is something that needs to be done before the previous task blocks:1
```

**Reply by primal-buddhist** [reply]
Many thanks, that looks like it might fit the bill.
